src/util/psbt/serialize.rs

```diff
@@ -144,3 +155,178 @@ impl Deserialize for SigHashType {
         }
     }
 }
+
+// Taproot related ser/deser
+impl Serialize for schnorr::PublicKey {
+    fn serialize(&self) -> Vec<u8> {
+        schnorr::PublicKey::serialize(&self).to_vec()
+    }
+}
+
+impl Deserialize for schnorr::PublicKey {
+    fn deserialize(bytes: &[u8]) -> Result<Self, encode::Error> {
+        schnorr::PublicKey::from_slice(bytes)
+            .map_err(|_| encode::Error::ParseFailed("Invalid xonly public key"))
+    }
+}
+
+impl Serialize for (schnorrsig::Signature, SchnorrSigHashType)  {
+    fn serialize(&self) -> Vec<u8> {
+        let mut ser_sig = self.0.as_ref().to_vec();
```

Comment:
Nit: here we also can do `Vec::with_capacity`

Reply:
I did not use because it could be 64 to 65 bytes depending on sighash type. And it felt awkward to write code for checking capacity difference with 1 byte. 

But now I see that can allocate it with 64 bytes for the most common case of sighash::default. And it won't be a big deal if it is 65 bytes. I personally don't feel like the extra lines of code for calculating 64vs 65 are worth it.